ChargeRateManager.h

```diff
@@ -0,0 +1,84 @@
+#ifndef CHARGERATEMANAGER
+#define CHARGERATEMANAGER
+
+class ChargeRateManager : public BatteryManagementSystem
+{
+    std::vector<Range> ChargeRateRange;
+    std::vector<float> ChargeRateLimit;
+    std::vector<std::string> messages;
+    typedef void (ChargeRateManager::*updateMessages)(void);
+    std::map<Language, updateMessages> languageMessagesList;
+    
+    public:
+    ChargeRateManager(bool withWarningLevel, Language messageLanguage)
+    {
+        populateChargeRateValues(withWarningLevel);
+        populateMessages(messageLanguage);
+    }
+    
+    void populateChargeRateValues(bool withWarningLevel)
+    {
+        if(withWarningLevel)
+        {
+            ChargeRateRange.push_back(NORMAL);
+            ChargeRateLimit.push_back(MAXIMUMCHARGERATE-(MAXIMUMCHARGERATE*TOLERANCE));
+            ChargeRateRange.push_back(HIGH_WARNING);
+            ChargeRateLimit.push_back(MAXIMUMCHARGERATE);
+            ChargeRateRange.push_back(HIGH_BREACH);
+        }
+        else
+        {
+            ChargeRateRange.push_back(NORMAL);
+            ChargeRateLimit.push_back(MAXIMUMCHARGERATE);
+            ChargeRateRange.push_back(HIGH_BREACH);
+        }
+    }
+    
+    void populateMessages(Language messageLanguage)
+    {
+        languageMessagesList.insert(std::pair<Language, updateMessages>(ENGLISH, &ChargeRateManager::populateEnglishMessages));
+        languageMessagesList.insert(std::pair<Language, updateMessages>(GERMAN, &ChargeRateManager::populateGermanMessages));
+        std::map<Language, updateMessages>::const_iterator messageIterator = languageMessagesList.find(messageLanguage);
+        assert(messageIterator != languageMessagesList.end());
+        (this->*messageIterator->second)();

```

Comment:
Understanding and getting confidence that `second` is correct requires some time, especially if the function gets bigger.
Why not use a struct instead of a pair?